GameTracker-Core/Controller.cs

```diff
@@ -0,0 +1,80 @@
+﻿using GameTracker_Core.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameTracker_Core
+{
+    public class Controller
+    {
+        private Device device { get; set; }
+
+        public Controller()
+        {
+            device = new Device();
+            if (!System.IO.Directory.Exists("C:\\Users\\Pete\\AppData\\Roaming\\GameTrackerAgent"))
+            {
+                System.IO.Directory.CreateDirectory("C:\\Users\\Pete\\AppData\\Roaming\\GameTrackerAgent");
+            }
+            device = Serializer.Load<Device>("C:\\Users\\Pete\\AppData\\Roaming\\GameTrackerAgent\\device.bin");
+        }
+
+        public void scanComputer()
+        {
+            //Find new or removed Games
+            List<GameDirectory> gameDirectories = device.GetGameDirectories();
+            List<Game> removedGames = new List<Game>();
+            List<Game> newGames = new List<Game>();
+            foreach(GameDirectory gameDirectory in gameDirectories)
+            {
+                removedGames.AddRange(DirectorySearchHelper.FindNotExistingGames(gameDirectory));
+                newGames.AddRange(DirectorySearchHelper.FindNewGames(gameDirectory));
```

Comment:
As discussed IRL, move the two loops below here

Reply:
on hold. I think there might be a problem. I look at this later.